autospec/specfiles.py

```diff
@@ -73,12 +75,40 @@ def __init__(self, url, version, name, release, config, requirements, content):
 
     def write_spec(self):
         """Write spec file."""
-        self.specfile = open_auto("{}/{}.spec".format(self.config.download_path, self.name), "w")
+        spec_path = f"{os.path.join(self.config.download_path, self.name)}.spec"
+        self.specfile = open_auto(spec_path, "w")
         self.specfile.write_strip = types.MethodType(_file_write, self.specfile)
 
+        # last change to sanitize url for template and build types
+        if self.config.urlban:
+            clean_url = re.sub(self.config.urlban, "localhost", self.url)
+            # Duplicate prefixes entry before we change the url
+            self.content.prefixes[clean_url] = self.content.prefixes.get(self.url)
+            self.url = clean_url
+
+        template_path = f"{spec_path}.template"
+
+        if os.path.isfile(template_path):
+            # make templates have a template build pattern
+            self.config.default_pattern = "template"
```

Comment:
Will there be a `write_template_pattern()` to implement this?

Reply:
Right now the second if isfile block is effectively that at the moment.

The template pattern at this point doesn't share anything outside the header with the normal flow though so I didn't want to make it seem too similar to the other patterns which are more focused on %build and %install sections.